third_party/amd/include/hipblas_instance.h

```diff
@@ -0,0 +1,277 @@
+    hipblasLtDestroy =
+        (hipblasLtDestroy_t)dlsym(dylibHandle, "hipblasLtDestroy");
+    hipblasLtMatmulDescCreate = (hipblasLtMatmulDescCreate_t)dlsym(
+        dylibHandle, "hipblasLtMatmulDescCreate");
+    hipblasLtMatmulDescDestroy = (hipblasLtMatmulDescDestroy_t)dlsym(
+        dylibHandle, "hipblasLtMatmulDescDestroy");
+    hipblasLtMatmulDescSetAttribute = (hipblasLtMatmulDescSetAttribute_t)dlsym(
+        dylibHandle, "hipblasLtMatmulDescSetAttribute");
+    hipblasLtMatrixLayoutCreate = (hipblasLtMatrixLayoutCreate_t)dlsym(
+        dylibHandle, "hipblasLtMatrixLayoutCreate");
+    hipblasLtMatrixLayoutDestroy = (hipblasLtMatrixLayoutDestroy_t)dlsym(
+        dylibHandle, "hipblasLtMatrixLayoutDestroy");
+    hipblasLtMatmulPreferenceCreate = (hipblasLtMatmulPreferenceCreate_t)dlsym(
+        dylibHandle, "hipblasLtMatmulPreferenceCreate");
+    hipblasLtMatmulPreferenceDestroy =
+        (hipblasLtMatmulPreferenceDestroy_t)dlsym(
+            dylibHandle, "hipblasLtMatmulPreferenceDestroy");
+    hipblasLtMatmulPreferenceSetAttribute =
+        (hipblasLtMatmulPreferenceSetAttribute_t)dlsym(
+            dylibHandle, "hipblasLtMatmulPreferenceSetAttribute");
+    hipblasLtMatmulAlgoGetHeuristic = (hipblasLtMatmulAlgoGetHeuristic_t)dlsym(
+        dylibHandle, "hipblasLtMatmulAlgoGetHeuristic");
+    hipblasLtMatmul = (hipblasLtMatmul_t)dlsym(dylibHandle, "hipblasLtMatmul");
+
+    const char *dlsym_error = dlerror();
+    if (dlsym_error) {
+      throw std::runtime_error("Could not load symbol from `" +
+                               std::string(name) +
+                               "`: " + std::string(dlsym_error));
+    }
+  }
+
+  void unloadHipBlasDylib() { dlclose(dylibHandle); }
+
+  void successOrExit(hipblasStatus_t status, const std::string &context = "") {
+    if (status != HIPBLAS_STATUS_SUCCESS) {
+      std::string error_msg =
+          "HIPBLAS Error in " + context + ": " + std::to_string(status);
+
+      switch (status) {
+      case HIPBLAS_STATUS_NOT_INITIALIZED:
+        error_msg += " (NOT_INITIALIZED)";
+        break;
+      case HIPBLAS_STATUS_ALLOC_FAILED:
+        error_msg += " (ALLOC_FAILED)";
+        break;
+      case HIPBLAS_STATUS_INVALID_VALUE:
+        error_msg += " (INVALID_VALUE - Parameters are unexpectedly NULL, in "
+                     "conflict or in impossible configuration)";
+        break;
+      case HIPBLAS_STATUS_MAPPING_ERROR:
+        error_msg += " (MAPPING_ERROR)";
+        break;
+      case HIPBLAS_STATUS_EXECUTION_FAILED:
+        error_msg += " (EXECUTION_FAILED)";
+        break;
+      case HIPBLAS_STATUS_INTERNAL_ERROR:
+        error_msg += " (INTERNAL_ERROR)";
+        break;
+      case HIPBLAS_STATUS_NOT_SUPPORTED:
+        error_msg += " (NOT_SUPPORTED)";
+        break;
+      case HIPBLAS_STATUS_INVALID_ENUM:
+        error_msg +=
+            " (INVALID_ENUM - unsupported enum value was passed to function)";
```

Comment:
Do we need the extra explanation? I feel the short symbol is already clear enough.